pkg/registry/core/service/ipallocator/allocator_test.go

```diff
@@ -166,18 +175,66 @@ func TestAllocateSmall(t *testing.T) {
 }
 
 func TestRangeSize(t *testing.T) {
-	testCases := map[string]int64{
-		"192.168.1.0/24": 256,
-		"192.168.1.0/32": 1,
-		"192.168.1.0/31": 2,
+	testCases := []struct {
+		name   string
+		cidr   string
+		addrs  int64
+		expect bool
+	}{
+		{
+			name:   "success IPv4 /24 cidr",
+			cidr:   "192.168.1.0/24",
+			addrs:  256,
+			expect: true,
+		},
+		{
+			name:   "failure IPv4 /24 cidr",
+			cidr:   "192.168.1.0/24",
+			addrs:  512,
+			expect: false,
+		},
+		{
+			name:   "failure unsupported IPv4 cidr",
+			cidr:   "192.168.1.0/1",
+			expect: false,
```

Comment:
In this test case, since tc.addrs == 0 and tc.expect == false, there's no check that RangeSize actually returns an error. So for example, if RangeSize is possibly changed in the future to not error for this CIDR, this test case would still pass. How about modifying the test like the following to try a mid-sized cidr, a max-sized cidr, and an oversized cidr (checking the error string for the latter):
```
func TestRangeSize(t *testing.T) {
	testCases := []struct {
		name   string
		cidr   string
		addrs  int64
		expErr string
	}{
		{
			name:  "success IPv4 /24 cidr",
			cidr:  "192.168.1.0/24",
			addrs: 256,
		},
		{
			name:  "success IPv4 /2 cidr",
			cidr:  "192.168.1.0/2",
			addrs: 1073741824,
		},
		{
			name:   "failure unsupported IPv4 cidr",
			cidr:   "192.168.1.0/1",
			expErr: "subnet mask must be longer",
		},
		{
			name:  "success IPv6 /98 cidr",
			cidr:  "2001:db8::/98",
			addrs: 1073741824,
		},
		{
			name:  "success IPv6 /66 cidr",
			cidr:  "2001:db8::/66",
			addrs: int64(1) << uint(62),
		},
		{
			name:   "failure unsupported IPv6 cidr",
			cidr:   "2001:db8::/65",
			expErr: "subnet mask must be longer",
		},
	}

	for _, tc := range testCases {
		_, cidr, err := net.ParseCIDR(tc.cidr)
		if err != nil {
			t.Fatal(err)
		}
		size, err := RangeSize(cidr)
		if err != nil {
			if tc.expErr == "" {
				t.Errorf("test %s failed, unexpected error: '%s'", tc.name, err.Error())
			} else if !strings.Contains(err.Error(), tc.expErr) {
				t.Errorf("test %s failed, expected '%s' error, got: '%s' error", tc.name, tc.expErr, err.Error())
			}
		} else {
			if tc.expErr == "" {
				if size != tc.addrs {
					t.Errorf("test %s failed. %s should have a range size of %d, got %d", tc.name, tc.cidr, tc.addrs, size)
				}
			} else {
				t.Errorf("test %s failed, expected error '%s' never occurred", tc.name, tc.expErr)
			}
		}
	}
}
```